### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by bmarty**
Improve the documentation of the project regarding default language.
Small cleanup
Add headers to SDK documentation. It will render like this:

image

(sadly I did not find a way to include links)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -124,7 +124,9 @@ As a general rule, please never edit or add or remove translations to the projec
 
 #### Adding new string
 
-When adding new string resources, please only add new entries in file `value/strings.xml`. Translations will be added later by the community of translators using Weblate.
+When adding new string resources, please only add new entries in the file `value/strings.xml`. Translations will be added later by the community of translators using Weblate.
+
+The file `value/strings.xml` must only contain American English (U. S. English) values, as this is the default language of the Android operating system. So for instance, please use "color" instead of "colour". Element Android will still use the language set on the system by the user, like any other Android applications which provide translations. The system language can be any other English language variants, or any other languages. Note that this is also possible to override the system language using the Element Android in-app language settings.
```
Comment (bmarty) [inline CONTRIBUTING.md:129]: @kittykat is it suitable to you?

File: matrix-sdk-android/build.gradle [hunk 2]
```diff
@@ -19,13 +19,18 @@ dokkaHtml {
         configureEach {
             // Emit warnings about not documented members.
             reportUndocumented.set(true)
-            // Suppress package legacy riot. Sadly this does not work
-            /*
+            // Suppress legacy Riot's packages.
             perPackageOption {
-                prefix.set("org.matrix.android.sdk.internal.legacy.riot")
+                matchingRegex.set("org.matrix.android.sdk.internal.legacy.riot")
```
Comment (bmarty) [inline matrix-sdk-android/build.gradle:24]: now it's working fine. Outdated documentation of dokka :/

File: matrix-sdk-android/docs/modules.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+## Welcome to the matrix-sdk-android documentation!
+
+This pages list the complete API that this SDK is exposing to a client application.
+
+*We are still building the documentation, so everything is not documented yet.*
+
+A few entry points:
+
+- **Matrix**: The app will have to create and manage a Matrix object.
```
Comment (bmarty) [inline matrix-sdk-android/docs/modules.md:11]: Could be nice to have link to the doc, but I did not find a way to make that work.

Comment (mnaturel) [reply matrix-sdk-android/docs/modules.md:11]: @bmarty That may help you to add the correct references to classes: https://github.com/Kotlin/dokka/issues/555#issuecomment-1111195409

File: CONTRIBUTING.md (hunk 1 above)
Comment (kittykat) [reply CONTRIBUTING.md:129]: Looks great, thanks!

**Review by kittykat — APPROVED** [review]

**Review by Claire1817 — APPROVED** [review]

File: matrix-sdk-android/docs/modules.md (hunk 3 above)
Comment (bmarty) [reply matrix-sdk-android/docs/modules.md:11]: Thanks, I will have a look on that!

**Pull request merged**